### PR TITLE
[Feature] Centralize error handling

### PR DESCRIPTION
**Pull request by chrisw000**
Invalid calls into paged results are currently not error handled correctly.

// make call with an invalid order status, to force a Bad Request 
var blowUp = await _client.OrdersService.GetAllOrdersAsync((OrderStatus)999));

This change centralizes the error handling an builds an exception with more information on it.

Where in the namespace would you want the IEndPoint? 

It would be useful for a later change to have SendServiceCall and SendHttpRequestMessagePagedAsync take in the IEndPoint as there parameter... and also have httpRequestMessageService.CreateHttpRequestMessage take in IEndPoint as it's parameter - this would allow the for the management of the request pipeline / prioritizing / rate limiting etc. based on the endpoint being called.

### CODE REVIEW

File: GDAXSharp.Specs/Services/Orders/OrdersServiceSpecs.cs [hunk 1]
```diff
@@ -171,9 +172,10 @@ class when_cancelling_order_by_id_with_failure
             Because of = () =>
                 exception = Catch.Exception(() => Subject.CancelOrderByIdAsync("144c6f8e-713f-4682-8435-5280fbe8b2b4").Result);
 
-            It should_have_correct_error_response_message = () =>
+            private It should_have_correct_error_response_message = () =>
```
Comment (dougdellolio) [inline GDAXSharp.Specs/Services/Orders/OrdersServiceSpecs.cs]: remove extra private :) RESHARPER!!

File: GDAXSharp/Services/IEndPoint.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Net.Http;
+
+namespace GDAXSharp.Services
+{
+    public interface IEndPoint
```
Comment (dougdellolio) [inline GDAXSharp/Services/IEndPoint.cs]: i don't have an opinion on where this should live. what do you think?

File: GDAXSharp/Services/IEndPoint.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        HttpMethod HttpMethod {get;}
+
+        string Uri {get;}
+
+        string Content { get; }
+    }
+
+    public class EndPoint : IEndPoint
+    {
+        public HttpMethod HttpMethod { get; }
```
Comment (dougdellolio) [inline GDAXSharp/Services/IEndPoint.cs]: nl these props and put below ctor

File: GDAXSharp/Services/AbstractService.cs [hunk 4]
```diff
@@ -44,14 +45,33 @@ public abstract class AbstractService
...
+                var jsonMsg = DeserializeObject<GDAXErrorMessage>(contentBody);
+                errorMessage = jsonMsg.Message;
+            }
+            catch
+            {
+                errorMessage = contentBody;
             }
 
-            return httpResponseMessage;
+            throw new GDAXSharpHttpException(errorMessage)
```
Comment (dougdellolio) [inline GDAXSharp/Services/AbstractService.cs]: 😎

File: GDAXSharp.Specs/Services/Orders/OrdersServiceSpecs.cs (hunk 1 above)
Comment (chrisw000) [reply GDAXSharp.Specs/Services/Orders/OrdersServiceSpecs.cs]: doh, every time :) yeah it automatically adds it

File: GDAXSharp/Services/IEndPoint.cs (hunk 2 above)
Comment (chrisw000) [reply GDAXSharp/Services/IEndPoint.cs]: In Network? Or just where it is for now? Happy to be guided by your naming conventions...

**Comment by chrisw000** [comment]
I've made those changes. 

I also noticed that `OrderService.CancelOrderByIdAsync(string id)` was the only place left skipping the error handling - so cleaned that up. It also seemed to be trying to handle a null response, which I don't think you'd get anyway? I took a guess that it was old code that used to work this way to handle not found? So I've added some error handling - not sure if this is wanted though?

Also removed some usings that should have been taken out on a previous pull request...

**Pull request merged**